OCPBUG 8125:Added Node Observability Agent (#69372)
openshift/openshift-docs: modules/node-observability-scripting-cr.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/using-node-observability-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="node-observability-scripting-cr_{context}"]
= Creating the Node Observability custom resource for scripting

> You must create and run the `NodeObservability` custom resource (CR) before you run the scripting. When you run the `NodeObservability` CR, it creates the necessary machine config and machine config pool CRs to enable the scripting on the worker nodes matching the `nodeSelector`.

Reviewer: Could you change this too something like (as it does not make use of machine config and machine config pools)

```
You must create and run the `NodeObservability` custom resource (CR) before you run the scripting. When you run the `NodeObservability` CR, it will enable the agent in scripting mode on the worker nodes matching the `nodeSelector`. 
```
```

Author: Right, so, I should rephrase it to explain there will be no machine config and machine config pods? Correct me if I am wrong, custom resource is to be created for scripting as well right?

Reviewer: > Right, so, I should rephrase it to explain there will be no machine config and machine config pods? Correct me if I am wrong, custom resource is to be created for scripting as well right?

Yes correct when you use the field "type" : "scripting" it does not make use of machine config and machine config pools. Yes we create a CR for both, the only difference is we use "type": "crio-kubelet" for CRIO profiling and "type": "scripting" to enable and execute the embedded scripts. Notice we can't do both at the same time, we will have to re-deploy to change from crio-kubelet to scripting and vice-versa

Author: Understood.

.Prerequisites
* You have installed the Node Observability Operator.
* You have installed the OpenShift CLI (oc).
* You have access to the cluster with `cluster-admin` privileges.

.Procedure

. Log in to the {product-title} CLI by running the following command:
+
[source,terminal]
----
$ oc login -u kubeadmin https://<HOSTNAME>:6443
----

. Switch back to the `node-observability-operator` namespace by running the following command:
+
[source,terminal]
----
$ oc project node-observability-operator
----

. Create a CR file named `nodeobservability.yaml` that contains the following text:
+
[source,yaml]
----
    apiVersion: nodeobservability.olm.openshift.io/v1alpha2
    kind: NodeObservability
    metadata:
      name: cluster <1>
    spec:
      nodeSelector:
        kubernetes.io/hostname: <node_hostname> <2>

>       type: scripting

Reviewer: type: scripting <3>

Author: Understood. And can you check the scripting info as well, that needs to be added here?

----
<1> You must specify the name as `cluster` because there should be only one `NodeObservability` CR per cluster.
<2> Specify the nodes on which the Node Observability agent must be deployed.

> 

Reviewer: <3>  type must be set to scripting (this will deploy the agent in scripting mode)

Author: Ack.

// Module included in the following assemblies:
//

> // * scalability_and_performance/using-node-observability-operator.adoc

Reviewer 2: ```suggestion
// * scalability_and_performance/node-observability-operator.adoc
```

Author: Ack.

:_mod-docs-content-type: PROCEDURE
[id="node-observability-scripting-cr_{context}"]
= Creating the Node Observability custom resource for scripting

> You must create and run the `NodeObservability` custom resource (CR) before you run the scripting. When you run the `NodeObservability` CR, it enables the agent in scripting mode on the compute nodes matching the `nodeSelector`. 

Author: Ack.

> You must create and run the `NodeObservability` custom resource (CR) before you run the scripting. When you run the `NodeObservability` CR, it will enable the agent in scripting mode on the worker nodes matching the `nodeSelector`. 

Author: Understood.

> You must create and run the `NodeObservability` custom resource (CR) before you run the scripting. When you run the `NodeObservability` CR, it will enable the agent in scripting mode on the compute nodes matching the `nodeSelector`. 

Reviewer 2: Suggestion to write the second sentence in present tense:
```suggestion
You must create and run the `NodeObservability` custom resource (CR) before you run the scripting. When you run the `NodeObservability` CR, it enables the agent in scripting mode on the compute nodes matching the `nodeSelector`. 
```

.Prerequisites
* You have installed the Node Observability Operator.

> * You have installed the OpenShift CLI (oc).

Author: Ack.

* You have access to the cluster with `cluster-admin` privileges.

.Procedure

> . Log in to the {product-title} CLI by running the following command:

Author: Ack

+
[source,terminal]
----

> $ oc login -u kubeadmin https://<HOSTNAME>:6443

Author: Ack.

----

> . Switch back to the `node-observability-operator` namespace by running the following command:

Author: Done

+
[source,terminal]
----
$ oc project node-observability-operator
----

> . Create a CR file named `nodeobservability.yaml` that contains the following text:

Author: Acknowledged

+
[source,yaml]
----
    apiVersion: nodeobservability.olm.openshift.io/v1alpha2
    kind: NodeObservability
    metadata:
      name: cluster <1>
    spec:
      nodeSelector:
        kubernetes.io/hostname: <node_hostname> <2>
      type: scripting <3>
----
<1> You must specify the name as `cluster` because there should be only one `NodeObservability` CR per cluster.
<2> Specify the nodes on which the Node Observability agent must be deployed.

> <3> To deploy the agent in scripting mode, you must set the type to scripting.

Author: Ack.

> . Run the `NodeObservability` CR:

Author: Will change.

+
[source,terminal]
----

> oc apply -f nodeobservability.yaml

Reviewer 2: Add the `$` here:
```suggestion
$ oc apply -f nodeobservability.yaml
```

Author: Ack.

----

+
.Example output
[source,terminal]
----
nodeobservability.olm.openshift.io/cluster created
----

. Review the status of the `NodeObservability` CR by running the following command:
+
[source,terminal]
----
$ oc get nob/cluster -o yaml | yq '.status.conditions'
----

+
.Example output
[source,terminal]
----
conditions:
  conditions:
  - lastTransitionTime: "2022-07-05T07:33:54Z"
    message: 'DaemonSet node-observability-ds ready: true NodeObservabilityScripting
      ready: true'
    reason: Ready
    status: "True"
    type: Ready
----

+

> `NodeObservability` CR run is completed when the reason is `Ready` and the status is `True`.

Author: Ack.

> `NodeObservability` CR run is completed when the `reason` is `Ready` and `status` is `"True"`.

Reviewer 2: Suggestion to start the sentence with "The":
```suggestion
The `NodeObservability` CR run is completed when the `reason` is `Ready` and `status` is `"True"`.
```

Author: Ack.
